extensions/positron-r/src/uri-handler.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+import { LOGGER } from './extension';
+import { RSessionManager } from './session-manager';
+import { EnvVar, RSession } from './session';
+
+export async function registerUriHandler() {
+	vscode.window.registerUriHandler({ handleUri });
+}
+
+// Example of a URI we expect to handle:
+// positron://positron.positron-r/cli?command=x-r-run:testthat::snapshot_review('snap')
+//
+// How the example URI breaks down:
+// {
+//    "scheme": "positron",
+//    "authority": "positron.positron-r",
+//    "path": "/cli",
+//    "query": "command=x-r-run:testthat::snapshot_review('zzz')",
+//    "fragment": "",
+//    "fsPath": "/cli"
+// }
+function handleUri(uri: vscode.Uri): void {
+	if (uri.path !== '/cli') {
+		return;
+	}
+
+	// Turns this query string
+	// "command=x-r-run:testthat::snapshot_review('zzz')"
+	// into this object
+	// { "command": "x-r-run:testthat::snapshot_review('zzz')" }
+	const query = new URLSearchParams(uri.query);
+	const command = query.get('command');
+	if (!command) {
+		return;
+	}
+
+	const commandRegex = /^(x-r-(help|run|vignette)):(.+)$/;
+	if (!commandRegex.test(command)) {
+		return;
+	}
+
+	const session = RSessionManager.instance.getConsoleSession();
+	if (!session) {
+		return;
+	}
+
+	session.openResource(command);
+	vscode.commands.executeCommand('workbench.panel.positronConsole.focus');
+}
+
+export async function prepCliEnvVars(session?: RSession): Promise<EnvVar> {
+	session = session || RSessionManager.instance.getConsoleSession();
+	if (!session) {
+		return {};
+	}
+
+	// cli 3.6.3.9001 gained support for configuring the URL format for hyperlinks.
+	const cliPkg = await session.packageVersion('cli', '3.6.3.9001');
+	const cliSupportsHyperlinks = cliPkg?.compatible ?? false;
+
+	if (!cliSupportsHyperlinks) {
+		// eslint-disable-next-line @typescript-eslint/naming-convention
+		return { R_CLI_HYPERLINKS: 'FALSE' };
+	}
```

Comment:
Here's where we exit early, without activating hyperlinks in the integrated terminal, in the absence of bleeding edge cli.